coordinate-systems.md

```diff
@@ -0,0 +1,246 @@
+
+Option (4) does exactly that - removes XRFrame and just assumes we are always talking about the current frame. All APIs are simplified, for example:
+
+
+```
+xrSession.setDevicePoseMode("stage");
+const devicePose = xrSession.getDevicePose();
+```
+
+
+Options (2), (3), and (4) all appear to be valid. Option (2) and (3) mainly differ in whether old XRFrame objects are invalidated or not. Option (3) removes some of the "don't do that" edge-case concerns, however it re-introduces the problem that XRFrame is meant to solve - that different platforms might not be able to handle calls dealing with world state between frames gracefully. Option (4) seems strictly better than (3) in that it at least simplifies the API surface greatly. So the question is really between (2) and (4) - 
+
+
+## Relative Hit-test API
+
+The problem of how to do pixel-perfect reticle functionality is a major use case of these APIs. The problem statement can be found [here](https://docs.google.com/document/d/1cGLkmihH4yZy5UlwyBSszWk0GGb8JJ4OGGI-INBX3YM/edit).
+
+In order to do a hit-test with a relative ray, we did an exploration of API shape given the current definitions/relationships of XRFrame, XRFOR, and XRCoordinateSystem. This is what we came up with:
+
+
+```
+XRSession.requestHitTest(XRRay ray, XRCoordinateSystem cs, (bool success, XRFrame frame, Array<XRHitResult> results) => {});
+XRHitResult.getMatrix(XRFrame, XRCoordinateSystem);
+```
+
+
+Let's go through each parameter and how it is used:
+
+*   input
+    *   ray: the ray to cast into the world and collect hits
+    *   coordinate system: represents the coordinate system to which the ray is relative. This is needed in order to be able to pass in a ray that is device-relative and get an accurate result for the future frame in which the calculation was made.
+*   callback
+    *   success: a boolean saying whether there was an error in the request. This is needed since we can no longer use a promise if we return an XRFrame, as XRFrames are not valid except within the callback where they are issued and thus, do to the ability to store a promise for later use, would be invalid in a late-bound promise.
```

Comment:
The reason the app would store the `XRFrame`-containing promise for a future frame would be to make use of that frame's pose data later when the promise is continued.  However, the alternate design proposed below also wouldn't enable that, since it does not provide an `XRFrame` either.  Coordinates in "absolute space" are also not usable beyond the frame in which they are returned, since "absolute space" would need to be able to drift as necessary to absorb head-tracking map updates.